### ISSUE TITLE
Create Recipes

### ISSUE THREAD

**Issue by DaveGerrardDMU** [DMU-CTEC2902/TeamTwistedFlax#3]
The website needs to allow authors to create recipes. As recipes are edited, it needs to keep a history of all the previous versions so we can see how the recipe changed.

**Comment by DaveGerrardDMU** [comment]
@NjoadAlghanim I've added a class diagram to the Wiki that neatens the original design up. I've taken the 'reciped change history' out for the time being, though, as I think recipe is quite complicated as it is (as it's the point where everyone else's work comes together). But, we can always put it back in later if you get this bit done in good time?

Read and finish all the labs, first, though... Then we can start properly after Easter. However, if you have any questions about this, add another comment here and @ me in it, and it'll send me an email. That way, all our discussion gets recorded nearer the code.

OK?

**Comment by NjoadAlghanim** [comment]
@DaveGerrardDMU  for the recipes class diagram I am going to add Preparation Time, Cooking Time and Serves for each recipe. is that ok??

**Comment by DaveGerrardDMU** [comment]
They all make sense to me. Make sure you create a Data Migration when you make any changes to the model, though (even just adding properties to the model classes). The data migration will create the code to add the extra columns to the database.

Do you know how to make a data migration?

**Comment by NjoadAlghanim** [comment]
yes from lab week 25

**Comment by DaveGerrardDMU** [comment]
Great. 

Don't forget to run Update-Database after you've created the migration, to apply the update.

Plus the design is just an outline. I don't mind if you do something different. As long as you push your changes into GitHub as soon as you make them. Make sure you push any data migrations you make into GitHub, too.

And have fun!

**Comment by AaronMulgrew** [comment]
@NjoadAlghanim Hi njoad, I am going to remove the preparation time, cooking time and server from the main index page, you still will be able to view them when you click on details? Is this ok, I can always change it back.
Thanks

**Comment by NjoadAlghanim** [comment]
Yes that's what I think about I'm going to change it today @AaronMulgrew

**Comment by AaronMulgrew** [comment]
@NjoadAlghanim I have just done it, have a look tell me if it should be different.
Thanks.

**Comment by AaronMulgrew** [comment]
@NjoadAlghanim  Hi njoad, I have uploaded some seed data to recipe, can you do some more, you can access it via the configuration file, you don't need to do a data migration once you have saved it, just type Update-Database.
